ParseCAN/spec/bus/frame.py

```diff
@@ -165,5 +189,36 @@ def __post_init__(self):
 
         self.frame.extend(frame)
 
-    def pack(self):
-        raise NotImplementedError()
+    def unpack(self, frame, **kwargs):
+        mux_id = frame[self.slice.start, self.slice.length]
+        return (self.name,
+                self.frame['key'][mux_id].unpack(frame, **kwargs))
+
+    def pack(self, id_tup, bitstring=0, length=None, by='name', **kwargs):
+        # get top level length
+        if length is None:
+            length = len(self)
+
+        name = id_tup[0]
```

Comment:
This solution is not very general, especially considering nested multiplexing. It's okay for now to assume we won't use that. We still need to at least implement a way to specify how to deeply pack things even if we don't use it. That way we won't be changing around the API later.